pkg/common/providers/ocmprovider/cluster.go

```diff
@@ -159,46 +159,43 @@ func (o *OCMProvider) LaunchCluster(clusterName string) (string, error) {
 				subnetIDs := strings.Split(viper.GetString(config.AWSVPCSubnetIDs), ",")
-				awsBuilder := v1.NewAWS().
-					AccountID(awsAccount).
-					AccessKeyID(awsAccessKey).
-					SecretAccessKey(awsSecretKey).
-					SubnetIDs(subnetIDs...)
-				newCluster = newCluster.CCS(v1.NewCCS().Enabled(true)).AWS(awsBuilder)
-				if len(subnetIDs) > 0 {
-					cloudProviderData, err := v1.NewCloudProviderData().
-						AWS(awsBuilder).
-						Region(v1.NewCloudRegion().ID(region)).
-						Build()
-					if err != nil {
-						return "", fmt.Errorf("error building AWS cloud provider data for retrieving Availability Zones: %v", err)
-					}
-					subnetworks, err := o.GetSubnetworks(cloudProviderData)
+				awsBuilder = awsBuilder.SubnetIDs(subnetIDs...)
+				cloudProviderData, err := v1.NewCloudProviderData().
+					AWS(awsBuilder).
+					Region(v1.NewCloudRegion().ID(region)).
+					Build()
+				if err != nil {
+					return "", fmt.Errorf("error building AWS cloud provider data for retrieving Availability Zones: %v", err)
+				}
+				subnetworks, err := o.GetSubnetworks(cloudProviderData)
+				if err != nil {
+					return "", fmt.Errorf("error retrieving AWS subnetworks: %v", err)
+				}
+				availabilityZones := GetAvailabilityZones(subnetworks, subnetIDs)
+				nodeBuilder.AvailabilityZones(availabilityZones...)
+			}
+			if viper.GetBool(config.Cluster.UseProxyForInstall) {
+				proxy := v1.NewProxy()
+				if userCaBundle := viper.GetString(config.Proxy.UserCABundle); userCaBundle != "" {
+					userCaBundleData, err := o.LoadUserCaBundleData(userCaBundle)
 					if err != nil {
-						return "", fmt.Errorf("error retrieving AWS subnetworks: %v", err)
+						return "", fmt.Errorf("error loading CA contents: %v", err)
 					}
-					availabilityZones := GetAvailabilityZones(subnetworks, subnetIDs)
-					nodeBuilder.AvailabilityZones(availabilityZones...)
+					newCluster = newCluster.AdditionalTrustBundle(userCaBundleData)
 				}
-				if viper.GetBool(config.Cluster.UseProxyForInstall) {
-					proxy := v1.NewProxy()
-					if userCaBundle := viper.GetString(config.Proxy.UserCABundle); userCaBundle != "" {
-						userCaBundleData, err := o.LoadUserCaBundleData(userCaBundle)
-						if err != nil {
-							return "", fmt.Errorf("error loading CA contents: %v", err)
-						}
-						newCluster = newCluster.AdditionalTrustBundle(userCaBundleData)
-					}
-					if httpProxy := viper.GetString(config.Proxy.HttpProxy); httpProxy != "" {
-						proxy = proxy.HTTPProxy(httpProxy)
-						newCluster = newCluster.Proxy(proxy)
-					}
-					if httpsProxy := viper.GetString(config.Proxy.HttpsProxy); httpsProxy != "" {
-						proxy = proxy.HTTPSProxy(httpsProxy)
-						newCluster = newCluster.Proxy(proxy)
-					}
+				if httpProxy := viper.GetString(config.Proxy.HttpProxy); httpProxy != "" {
+					proxy = proxy.HTTPProxy(httpProxy)
+					newCluster = newCluster.Proxy(proxy)
+				}
+				if httpsProxy := viper.GetString(config.Proxy.HttpsProxy); httpsProxy != "" {
```

Comment:
I think we would only want the httpsProxy if there is a CA bundle, right?

Reply:
The PR moved it out of an if block but that http proxy logic is untouched. 
It does check for the CA Bundle over here:
https://github.com/openshift/osde2e/pull/1520/files#diff-360361bd516bd9b8a5a85b5a4ce53db8fa5e5682353cf4791c44a6aaeb2f7eadR185